### ISSUE TITLE
Checking for duplicate phone numbers while adding contacts in Circle of Trust does not ignore special characters

### ISSUE THREAD

**Issue by Pooja-Mahadev-Soundalgekar** [systers/PCSA-Android#420]
In Circle of Trust, while adding a contact, the algorithm for checking if its a duplicate phone number does not ignore white spaces and hyphens that are usually part of the phone number.

For example : +1XXXXXXXXXX, +1 XXX XXX XXXX and +1-XXX-XXX-XXXX should be detected as the same number. The video below shows this bug. It only detects duplicate phone numbers if its the exact same string. 

![duplicatecontact](https://cloud.githubusercontent.com/assets/12373978/23755370/230e425c-0506-11e7-90c5-135a24b52507.gif)

@sandarumk I would like to work on this bug.

**Comment by HimanshuS01** [comment]
hey @Pooja-Mahadev-Soundalgekar This issue is a duplicate of issue no #214 . You can close this issue .
_Thank you_  :)

**Comment by Pooja-Mahadev-Soundalgekar** [comment]
@HimanshuS01 The issue you have mentioned is not the same as this one. This issue arises because the format of the phone numbers are not taken into account in the logic for checking of duplicate numbers.

**Comment by HimanshuS01** [comment]
@Pooja-Mahadev-Soundalgekar I think issue  number #158 and #193 are somehow  related to the **formatting problem**(spaces and hypens in your case) of the phone numbers ,while adding contacts either from the phonebook or entering manually. 
as in the issue no #193 @rohands also talked about the failure in the code logic to check duplicacy in this case.
However Its better to wait for the  @sandarumk 's reply in this issue .
Thanks :)

**Comment by Pooja-Mahadev-Soundalgekar** [comment]
@HimanshuS01 Yes those issues are more or less related to this issue so I guess solving one will solve the others too. Thanks for pointing it out :)

**Comment by sandarumk** [comment]
Will come to this validating phone number issue after this hassle is finished. Will keep it open and onhold.
